### ISSUE TITLE
There are no option to use USE_DLFCN from qt input plugin

### ISSUE THREAD

**Issue by ilya-fedin** [hamonikr/nimf#20]
I'm trying to get linking of input plugins in tdesktop without patching of Qt. Currently it uses [a way old nimf](https://github.com/desktop-app/nimf) and integrates it [using a Qt patch](https://github.com/desktop-app/patches/blob/master/qtbase_5_12_8.diff#L504). I just built a plugin successfully, [but with some patching](https://github.com/ilya-fedin/kotatogram-desktop/blob/add-nimf-to-appimage/Telegram/Patches/nimf.diff). And even after that it requires a couple of libraries that isn't needed in fact.

TL;DR of what is needed
* Checking qt plugin path in qt_prefix/plugins (ideally - an option for a builder to specify it)
* An option to build and install only qt input plugin
* An option to build qt input plugin with USE_DLFCN and, therefore, not requesting dependencies for libnimf and other things (libhangul, libxklavier, libappindicator, etc)
* An option to build qt input plugin statically (to not to replace shared_library with static_library with sed and to not to add -DQT_STATICPLUGIN to CXXFLAGS)

**Comment by chaeya** [comment]
Thank you for your contribution to better nimf
I would like to apply your patch and test it first, then keep talking.

**Comment by chaeya** [comment]
@ilya-fedin I can't find the patch file in the link. Please check it again.

**Comment by ilya-fedin** [comment]
That's the new link: https://github.com/kotatogram/kotatogram-desktop/blob/dev/Telegram/Patches/nimf.diff
But it was the most simple way to do compilation for me, I think you can do it with meson options :)

**Comment by chaeya** [comment]
@ilya-fedin Thank you for patch. I confirmed there was no problem after applying your patch and building.
